pkg/build/builder/cmd/scmauth/password.go

```diff
@@ -0,0 +1,107 @@
+package scmauth
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	DefaultUsername      = "builder"
+	UsernamePasswordName = "password"
+	UsernameSecret       = "username"
+	PasswordSecret       = "password"
+	TokenSecret          = "token"
+	UserPassGitConfig    = `# username and password git config
+[url "%[1]s"]
+   insteadOf = %[2]s
+`
+)
+
+// SSHPrivateKey implements SCMAuth interface for using SSH private keys.
+type UsernamePassword struct {
+	SourceURL url.URL
+	SAToken   url.URL
+}
+
+// Setup creates a wrapper script for SSH command to be able to use the provided
+// SSH key while accessing private repository.
+func (u UsernamePassword) Setup(baseDir string) error {
+
+	// Only apply to https and http URLs
+	if scheme := strings.ToLower(u.SourceURL.Scheme); scheme != "http" && scheme != "https" {
+		return nil
+	}
+
+	// Determine username
+	// 1. Look for a username secret
+	username, err := readSecret(baseDir, UsernameSecret)
+	if err != nil {
+		return err
+	}
+	// 2. If not provided, look at the username in the URL
+	if username == "" && u.SourceURL.User != nil {
+		username = u.SourceURL.User.Username()
+	}
+	// 3. If still not found, use a default username
+	if username == "" {
+		username = DefaultUsername
+	}
+
+	// Determine password
+	// 1. Look for a token secret
+	password, err := readSecret(baseDir, TokenSecret)
+	if err != nil {
+		return err
+	}
+	// 2. Look for a password secret
+	if password == "" {
+		password, err = readSecret(baseDir, PasswordSecret)
+		if err != nil {
+			return err
+		}
+	}
+	// 3. Look for a password in the URL
+	if password == "" && u.SourceURL.User != nil {
+		password, _ = u.SourceURL.User.Password()
+	}
+
+	gitconfig, err := ioutil.TempFile("", "gituserpass.")
+	if err != nil {
+		return err
+	}
+	defer gitconfig.Close()
+
+	usernamePasswordURL := u.SourceURL
+	usernamePasswordURL.User = url.UserPassword(username, password)
+	gitconfig.WriteString(fmt.Sprintf(UserPassGitConfig, usernamePasswordURL.String(), u.SourceURL.String()))
```

Comment:
It doesn't because we don't output the stdout/stderr of the git clone command. However, when invoking git directly from the shell it does show up in some cases.